lib/lint.ml

```diff
@@ -88,10 +88,14 @@ let lint_res ~msgf = function
 let pp_field = Fmt.(styled `Bold string)
 
 let lint_opam_doc pkg =
-  lint_res
-    ~msgf:(fun l ->
-      l "opam field %a can be parsed by dune-release" pp_field "doc")
-    (Pkg.doc_user_repo_and_path pkg)
+  ( match Pkg.doc_user_repo_and_path pkg with
+  | Ok _ ->
+      report_status `Ok (fun l ->
+          l "opam field %a can be parsed by dune-release" pp_field "doc")
+  | Error _ ->
```

Comment:
If we're going to allow missing doc fields, how about not reporting a failure (even though it's non fatal) if it's missing?

What would you think about the following:
- If there's no doc field, we mention it with a `Skipping doc field linting, no doc field for package xxx` or something similar, and succeed
- If there's one, we report whether we can parse it as you did there, in a non-fatal way so that it doesn't hurt people hosting there doc on readthedocs for instance.